internal/fleet/agent_policy/models.go

```diff
@@ -851,6 +899,213 @@ func mergeAgentFeature(existing []apiAgentFeature, newFeature *apiAgentFeature)
 	return &result
 }
 
+// advancedSettingsAttrTypes returns attribute types for advanced_settings pulled from the schema
+func advancedSettingsAttrTypes() map[string]attr.Type {
+	return getSchema().Attributes["advanced_settings"].GetType().(attr.TypeWithAttributeTypes).AttributeTypes()
+}
+
+// populateAdvancedSettingsFromAPI populates the advanced settings from API response
+func (model *agentPolicyModel) populateAdvancedSettingsFromAPI(ctx context.Context, data *kbapi.AgentPolicy) diag.Diagnostics {
+	if data.AdvancedSettings == nil {
+		model.AdvancedSettings = types.ObjectNull(advancedSettingsAttrTypes())
+		return nil
+	}
+
+	settings := advancedSettingsModel{}
+
+	// Logging level
+	if data.AdvancedSettings.AgentLoggingLevel != nil {
```

Comment:
Can we open an issue in the Kibana repo to fixup the typing in the openapi spec for these fields.

Reply:
https://github.com/elastic/kibana/issues/246372